### PR TITLE
removes decord 

### PR DESCRIPTION
**Pull request by vrnvu**
The only usage that was left of decord is removed in this PR

- Context:, project abandoned for 2 years: https://github.com/dmlc/decord
- Adds toil to local dev (https://github.com/huggingface/transformers/issues/22617)
  - https://github.com/huggingface/transformers/issues?q=is%3Aissue+decord
- Already removed from a couple of places in favor of av: https://github.com/huggingface/transformers/pulls?q=is%3Apr+decord

The helper functions `read_video_pyav` and `sample_frame_indices` are the used in all the documentation https://github.com/search?q=repo%3Ahuggingface%2Ftransformers%20read_video_pyav&type=code and `video_classification` https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/video_classification.py#L132

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
Very welcome for this! Thanks for the cleanup 🤗 
Could you confirm that the `test_pipeline_video_classification` tests work well locally?

**Comment by ArthurZucker** [comment]
Regarding the failing test: 
```python
VideoClassificationPipeline requires the PyAv library but it was not found in your environment. You can install it with:
```
pip install av
```
Please note that you may need to restart your runtime after installation.
FAILED tests/models/timesformer/test_modeling_timesformer.py::TimesformerModelTest::test_pipeline_video_classification_fp16 - ImportError: 
VideoClassificationPipeline requires the PyAv library but it was not found in your environment. You can install it with:
```
pip install av
```
Please note that you may need to restart your runtime after installation.
```
the workflow for pipelines needs a small fix!

**Comment by vrnvu** [comment]
Fixed the requirement, I wrongly confused "vision" with "video" and thought I didn't need the `av` requirement. 🤕 

The other 2 test failing seem flaky test right? they have nothing to do with this MR.

**Review by ArthurZucker — APPROVED** [review]
Thanks, yep the other tests are unrelated!

**Comment by ArthurZucker** [comment]
But the tests you edited are skipped by the CIs because they need av (which was not installed)

File: src/transformers/models/git/convert_git_to_pytorch.py [hunk 1]
```diff
@@ -193,10 +194,27 @@ def prepare_img(model_name):
...
+        """
+        frames = []
+        container.seek(0)
+        start_index = indices[0]
+        end_index = indices[-1]
+        for i, frame in enumerate(container.decode(video=0)):
+            if i > end_index:
+                break
+            if i >= start_index and i in indices:
+                frames.append(frame)
```
Comment (ArthurZucker) [inline src/transformers/models/git/convert_git_to_pytorch.py:216]:
```suggestion
                frames.append(frame.to_rgb())
```
any reason we don't use this? 
or directly convert before stacking?

**Review by ArthurZucker — APPROVED** [review]
Small nit, LGTM otherwise!

File: src/transformers/models/git/convert_git_to_pytorch.py (hunk 1 above)
Comment (vrnvu) [reply src/transformers/models/git/convert_git_to_pytorch.py:216]: Good question, we return the transformation at the end to directly transform the numpy data structure

```
return np.stack([x.to_ndarray(format="rgb24") for x in frames])
```

It's more efficient than converting the frames to rgb:
https://github.com/PyAV-Org/PyAV/blob/main/av/video/frame.pyx#L252
https://github.com/PyAV-Org/PyAV/blob/main/av/audio/frame.pyx#L168

**Comment by ArthurZucker** [comment]
Okay merging then! 🤗

**Pull request merged**